### PR TITLE
Editing pass on recsys

### PR DESCRIPTION
**Pull request by henryre**
Just some LF formatting stuff and clarity

### CODE REVIEW

File: recsys/recsys_tutorial.py [hunk 1]
```diff
@@ -1,6 +1,19 @@
...
+# In this tutorial, we'll provide a simple walkthrough of how to use Snorkel to build a recommender system.
+# We consider a setting similar to the [Netflix challenge](https://www.kaggle.com/netflix-inc/netflix-prize-data), but with books instead of movies.
+# We have a set of users and books, and for each user we know the set of books they have interacted with (read or marked as to-read).
+# We don't have the user's numerical ratings for the books they read, except in a small number of cases.
+# We also have some text reviews written by users.
+#
+# Our goal is to build a recommender system by training a classifier to predict whether a user will read and like any given book.
+# We represent users using the set of books they have interacted with, and train a model over a user-book pair to predict a `rating` (a `rating` of 1 means the user will read and like the book).
+# We represent a user by the set of books they interacted with.
```
Comment (brahmaneya) [inline recsys/recsys_tutorial.py]: This sentence is repeated

File: recsys/recsys_tutorial.py [hunk 2]
```diff
@@ -111,7 +134,7 @@ def stars_in_review(x):
 
 @preprocessor(memoize=True)
 def textblob_polarity(x):
-    if x.review_text:
+    if isinstance(x.review_text, str):
         x.blob = TextBlob(str(x.review_text))
```
Comment (brahmaneya) [inline recsys/recsys_tutorial.py]: Don't need the str cast. Actually, I don't remember why I had that cast in the first place, since non-None review_text's should always be strings

File: recsys/recsys_tutorial.py [hunk 3]
```diff
@@ -164,11 +187,11 @@ def polarity_negative(x):
 # ### Applying labeling functions to the training set
 #
 # We apply the labeling functions to the training set, and then filter out examples unlabeled by any LF, and combine the rest with the dev set to form our final training set.
+# We choose to train on our dev set in this case because it's relatively large (85.3k examples) and we didn't overfit to it when writing our LFs.
```
Comment (brahmaneya) [inline recsys/recsys_tutorial.py]: In general, it should always be ok to train on the dev set. Even if you overfit, as long as the final labels are correct, the end model should benefit from it.
Note that we are also using dev set for training (along with labeled part of the train set) in the spouse tutorial.

File: recsys/recsys_tutorial.py (hunk 2 above)
Comment (henryre) [reply recsys/recsys_tutorial.py]: One of them is the integer 70

File: recsys/recsys_tutorial.py [hunk 4]
```diff
@@ -163,45 +198,47 @@ def polarity_negative(x):
...
+# This type of model is common in large-scale recommender systems, for example, the [YouTube recommender system](https://ai.google/research/pubs/pub45530).
 
 # %%
 import numpy as np
 import tensorflow as tf
-from utils import precision, recall, f1
+from utils import precision_batch, recall_batch, f1_batch
 
-n_books = len(df_books)
+n_books = max([max(df.book_idx) for df in [df_train, df_test, df_dev, df_valid]])
```
Comment (henryre) [inline recsys/recsys_tutorial.py]: good catch!

File: recsys/utils.py [hunk 5]
```diff
@@ -1,47 +1,92 @@
 import calendar
+from datetime import datetime
```
Comment (henryre) [inline recsys/utils.py]: nit: from... after import...

File: recsys/utils.py [hunk 6]
```diff
@@ -1,47 +1,92 @@
...
 import gzip
 import json
+import logging
 import os
+from typing import Any, Dict, List, Optional, Tuple
+
+import numpy as np
 import pandas as pd
-from datetime import datetime
+import pickle
```
Comment (henryre) [inline recsys/utils.py]: nit: std lib

File: recsys/utils.py [hunk 7]
```diff
@@ -1,47 +1,92 @@
...
 from sklearn.model_selection import train_test_split
 from tensorflow.keras import backend as K
 
+logging.basicConfig(level=logging.INFO)
```
Comment (henryre) [inline recsys/utils.py]: nit: rm

File: recsys/utils.py [hunk 8]
```diff
@@ -1,47 +1,92 @@
...
+
+YA_BOOKS_URL = "https://drive.google.com/uc?id=1gH7dG4yQzZykTpbHYsrw2nFknjUm0Mol"
+YA_INTERACTIONS_URL = "https://drive.google.com/uc?id=1NNX7SWcKahezLFNyiW88QFPAqOAYP5qg"
+YA_REVIEWS_URL = "https://drive.google.com/uc?id=1M5iqCZ8a7rZRtsmY5KQ5rYnP9S0bQJVo"
+SMALL_DATA_URL = "https://drive.google.com/uc?id=1_UY4xTbk3o0xjGbVllQZC2bBt-WAwyF_"
+
+BOOK_DATA = "data/goodreads_books_young_adult.json.gz"
+INTERACTIONS_DATA = "data/goodreads_interactions_young_adult.json.gz"
+REVIEWS_DATA = "data/goodreads_reviews_young_adult.json.gz"
+SAMPLE_DATA = "data/sample_data.pkl"
```
Comment (henryre) [inline recsys/utils.py]: hmmmm, can we go back to the smaller json.gz versions? more authentic test of the data utils

File: tox.ini [hunk 9]
```diff
@@ -20,7 +21,9 @@ deps =
     crowdsourcing: -rcrowdsourcing/requirements.txt
     recsys: -rrecsys/requirements.txt
     drybell: -rdrybell/requirements.txt
-passenv = JAVA_HOME
+passenv =
+    JAVA_HOME
+    TRAVIS
```
Comment (henryre) [inline tox.ini]: 😭

File: recsys/utils.py (hunk 8 above)
Comment (brahmaneya) [reply recsys/utils.py]: The trouble with that is, It's hard to get books, reviews, and interactions that have high enough intersection with each other. If we just sample ~ 1000 of each, we end up with none of the dev interactions having reviews, most interactions having no book metadata, and so on. So LFs get 0 coverage, and loss becomes nan.

File: recsys/utils.py (hunk 6 above)
Comment (brahmaneya) [reply recsys/utils.py]: Ran isort -rc on this file.

**Comment by brahmaneya** [comment]
Submitting because the failed test is unrelated (spam slicing)

**Comment by henryre** [comment]
:+1: let's get this in
